src/bots/teams/src/index.ts

```diff
@@ -53,16 +53,30 @@ const file = fs.createWriteStream(__dirname + "/test.webm");
   const stream = await getStream(page, { audio: true, video: true });
 
   console.log("Recording...");
-
   stream.pipe(file);
-  setTimeout(async () => {
-    await stream.destroy();
-    file.close();
-    console.log("finished");
-
-    await browser.close();
-    (await wss).close();
-  }, 1000 * 10);
 
-  // await browser.close();
+  // First wait for the leave button to appear (meaning we've joined the meeting)
+  await page.waitForSelector('button[aria-label="Leave (Ctrl+Shift+H)"]', {
+    timeout: 30000,
+  });
+  console.log("Successfully joined meeting");
+
+  // Then wait for meeting to end by watching for the "Leave" button to disappear
```

Comment:
I don't think this handles the case when a meeting is ended for all participants / the bot is kicked by a user, right? Maybe we should make a note of that to handle in the future

Reply:
I think it should because if the bot is kicked then the leave button will disappear?

Reply:
@owengretzinger Yes this is true!